File: cli/src/output.test.ts
import { describe, expect, test } from 'bun:test';
import { expandOutput } from './output.js';

describe('expandOutput', () => {
  test('expands placeholders', () => {
    expect(
      expandOutput('i18n/{lang}/{relpath}', {
        lang: 'es',
        relpath: 'docs/guide.md',
        basename: 'guide',
        ext: 'md',
      }),
    ).toBe('i18n/es/docs/guide.md');
  });

  test('normalizes slashes', () => {
    expect(
      expandOutput('out\\{lang}\\{basename}.{ext}', {
        lang: 'de',
        relpath: 'docs\\guide.md',
        basename: 'guide',
        ext: 'md',
      }),
    ).toBe('out/de/guide.md');
  });
});

File: cli/internal/glossia/output_test.go
package glossia

import "testing"

func TestExpandOutput_ExpandsPlaceholders(t *testing.T) {
	got := expandOutput("i18n/{lang}/{relpath}", OutputValues{
		Lang:     "es",
		RelPath:  "docs/guide.md",
		Basename: "guide",
		Ext:      "md",
	})
	if got != "i18n/es/docs/guide.md" {
		t.Fatalf("unexpected output: %q", got)
	}
}

func TestExpandOutput_NormalizesSlashes(t *testing.T) {
	got := expandOutput("out\\{lang}\\{basename}.{ext}", OutputValues{
		Lang:     "de",
		RelPath:  "docs\\guide.md",
		Basename: "guide",
		Ext:      "md",
	})
	if got != "out/de/guide.md" {
		t.Fatalf("unexpected output: %q", got)
	}
}
